### DISCUSSION TITLE
Configuring hitRadius for annotations

### DISCUSSION THREAD

**Discussion by aafanaseva-shell** [chartjs/chartjs-plugin-annotation#899] [Q&A]
Hello!

In chart js therre is a possibility to configure `hitRadius` for data points which works like making "hitboxes" bigger. Is there anything similar for annotations in this plugin? 
I didn't find any option that can solve this issue - I understand that I probably can create custom `interaction mode` or override global onHover handler for Chart object, but it looks like there probably should be simple way to manage this through options like it's done in Chart js?

(line charts, line annotations)

**Comment by stockiNail** [comment]
@aafanasev-shell just to avoid misunderstanding, are you looking for a way to change an annotation when hovering? for instance, in your case, using a line annotation, to increase the borderWidth in order to show it bigger?

**Reply by aafanaseva-shell** [reply]
Yep, in my case I have a chart with lot's of different horizontal line annotations. And since there are quite a bit of them, I cannot make them thick enough. 
So, the goal I'm trying to achieve is make "active" area for annotation bigger, while not making it's border thicker. Smth similar to button's hitboxes in regular GUIs 🙂

**Reply by stockiNail** [reply]
OK!, That's not implemented yet, and as you mentioned, you could do it now implementing a custom interaction or playing with `onHover` at chart level. But maybe it could be a enhancement.

**Reply by aafanaseva-shell** [reply]
Got it! 
Thanks for quick response! 🙂

**Reply by stockiNail** [reply]
@aafanasev-shell FYI I have created an issue in order do not forget it: https://github.com/chartjs/chartjs-plugin-annotation/issues/900

**Reply by stockiNail** [reply]
@aafanasev-shell I thought a bit how to implement it.

First of all, I think `hitRadius` option name could be misleading (radius is usually related to a circle or curve) therefore I was thinking to use another name, for instance `hitTolerance`.

The property is the amount of pixels to use to understand if the mouse points on the annotation. Default is 0 (to avoid any breaking change).

The issue that I see could be on line annotation when is defined as a curve, where we are using a Path2D object to know if the mouse point is on the line.I hope to find out a good solution.

If you agree, I'll change the line annotation as first and (before submitting the PR) I would like to prepare a code pen for you, where you can play with it and to give me feedbacks if the implementation is fitting with the need.
WDYT?

FYI: I don't know now when I can start doing it because busy on other stuff.

**Reply by aafanaseva-shell** [reply]
The solution you're proposing sounds great and I'll be glad to check the codepen 🙂 
so far I was able to find a workaround which still provide good enough UX: when the cursor is over annotation (`enter` event), I increase `borderWidth` of annotation, move it on top of others (simple reordering), and before all of these I also check if there is any other annotation already "highlighted" this way to prevent "jumping" of selection between different annotations. 
Although it doesn't make it easier to "catch" the annotation, it reduces the risk of accidentally triggering `leave` event, which overall provides almost the same UX as if we already had larger `hitTolerance` 🙂

**Reply by stockiNail** [reply]
@aafanasev-shell here is a prototype on line annotation: https://codepen.io/stockinail/pen/oNQoyYg

**Reply by aafanaseva-shell** [reply]
works like a charm! 🙂 I also checked `click` handler and it also worked 👍

**Reply by stockiNail** [reply]
Thank you very much for testing and feedback.
Now I need to implement for all annotations, if make sense. Let me take time to do it.
I need of maintainer approval

**Reply by stockiNail** [reply]
@aafanasev-shell FYI https://github.com/chartjs/chartjs-plugin-annotation/pull/902

**Reply by aafanaseva-shell** [reply]
Great, you're super fast! 🙂

**Reply by stockiNail** [reply]
Unfortunately I have to manage other stuff now. I'll come back soon to complete the PR. I hope to finish tomorrow.

**Reply by stockiNail** [reply]
PR has been completed and opened for review by maintainers
